### DISCUSSION TITLE
Sample custom_config.h: www back/forward

### DISCUSSION THREAD

**Discussion by idavydov** [manna-harbour/miryoku#134] [Show and tell]
Hi,
Since there were no examples, I'm sharing my very simple customization. I'm using vi navigation and since I'm not using RGB LEDs, I rebind `Y` and `O` (for QWERTY) to browser's back and forward respectively.

I just copied `MIRYOKU_ALTERNATIVES_MEDIA_VI` to `custom_config.h`, renamed it to `MIRYOKU_LAYER_MEDIA` and replaced some keycodes:

```
#define MIRYOKU_LAYER_MEDIA \
RESET,             U_NA,              U_NA,              U_NA,              U_NA,              KC_WBAK,           RGB_MOD,           RGB_HUI,           KC_WFWD,           RGB_VAI,           \
KC_LGUI,           KC_LALT,           KC_LCTL,           KC_LSFT,           U_NA,              KC_MPRV,           KC_VOLD,           KC_VOLU,           KC_MNXT,           OUT_AUTO,          \
U_NA,              KC_ALGR,           U_NA,              U_NA,              U_NA,              U_NU,              U_NU,              U_NU,              U_NU,              U_NU,              \
U_NP,              U_NP,              U_NA,              U_NA,              U_NA,              KC_MSTP,           KC_MPLY,           KC_MUTE,           U_NP,              U_NP
```

![Screenshot from 2022-08-27 21-23-36](https://user-images.githubusercontent.com/671660/187061769-79354bd0-082d-4574-99f3-8680ccc06c03.png)

I'm not sure if that would also work with bluetooth profile keys, since for all of them the keycode is the same.

Additionally I had to take `users/manna-harbour_miryoku/miryoku_babel` from the `miryoku` branch:
```
git remote add miryoku_qmk https://github.com/manna-harbour/miryoku_qmk.git
git fetch miryoku_qmk
git checout miryoku -- users/manna-harbour_miryoku/miryoku_babel
```

**Comment by manna-harbour** [comment]
Thanks for being the first to share your config!

The BT profile selection keys aren't currently used in Miryoku QMK as QMK lacks that feature, so those positions are all available for use here.

For Miryoku QMK, use of `custom_config.h` currently requires the Miryoku QMK development branch.

**Comment by Daveyr** [comment]
I've not been able to implement this so far. Where are you editing the custom_config.h? I'm using the blank one in `qmk_firmware/users/manna-harbour_miryoku/custom_firmware.h` and modifying the base layer using `MIRYOKU_LAYER` as per the docs.

I'm also using a community layout (ortho4x8), extended thumbs, invertedT, etc. Does that prevent this from working?

**Reply by manna-harbour** [reply]
@Daveyr Unless you are commenting on this particular custom config shared in this Discussion, please create a new discussion.

That's the correct location. See near the top of the *custom_config.h* section at https://github.com/manna-harbour/miryoku/discussions/85 to confirm the path.

To substitute the Base layer, define `MIRYOKU_LAYER_BASE`.

Community layouts are supported in the same way. 

If you substitute a layer it will override any configuration options for that layer. So if you substitute a custom Base, you can still choose alternative mapping options (extended thumbs) and alternative Nav options (inverted T), but you can't choose alternative Base layer options.
